modules/storage/fsx-lustre/stack.py

```diff
@@ -0,0 +1,116 @@
+#  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License").
+#    You may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+
+import logging
+from typing import Any, List, Optional, Union, cast
+
+import aws_cdk.aws_ec2 as ec2
+import aws_cdk.aws_fsx as fsx
+from aws_cdk import Aspects, CfnTag, Stack, Tags
+from cdk_nag import AwsSolutionsChecks, NagSuppressions
+from constructs import Construct, IConstruct
+
+_logger: logging.Logger = logging.getLogger(__name__)
+
+
+class FsxFileSystem(Stack):
+    def __init__(
+        self,
+        scope: Construct,
+        id: str,
+        *,
+        project_name: str,
+        deployment_name: str,
+        module_name: str,
+        vpc_id: str,
+        fs_deployment_type: str,
+        private_subnet_ids: List[str],
+        storage_throughput: Union[int, float, None],
+        data_bucket_name: Optional[str],
+        export_path: Optional[str],
+        import_path: Optional[str],
+        **kwargs: Any,
+    ) -> None:
+        # Env vars
+        self.project_name = project_name
+        self.deployment_name = deployment_name
+        self.module_name = module_name
+
+        super().__init__(scope, id, **kwargs)
+        Tags.of(scope=cast(IConstruct, self)).add(key="Deployment", value=f"{project_name}-{deployment_name}")
```

Comment:
Can you add the below implementation for tagging, so that the entire name of project, dep and module name can be propagate ?

```python
# used to tag AWS resources. Tag Value length cant exceed 256 characters
 full_dep_mod = dep_mod[:256] if len(dep_mod) > 256 else dep_mod
# Tagging all resources
Tags.of(scope=cast(IConstruct, self)).add(key="Deployment", value=full_dep_mod)
```

Reply:
yes